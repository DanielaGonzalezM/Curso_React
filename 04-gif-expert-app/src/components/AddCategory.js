import React,{useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setcategories}) => {

const [inputValue, setInputValue] = useState('');
const handleInputChange=(e)=>{
    setInputValue(e.target.value);
}
const handleSubmit=(e)=>{
    e.preventDefault();
    if(inputValue.trim().length>2){
        setcategories(cats=>[...cats,inputValue]);
        setInputValue('');
    }
    

}
    return (
   
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            
            />
</form>
        
    )
}


AddCategory.propTypes = {
    setcategories: PropTypes.func.isRequired
}