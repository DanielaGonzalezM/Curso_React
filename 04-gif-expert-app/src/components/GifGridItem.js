import React from 'react';
import { PropTypes } from 'prop-types';

export const GifGridItem = ({title,url}) => {
    
  return (
    <div className='card  animate__animated  animate__zoomIn'>
    
        <img src={url} alt={title}></img>
        <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes={
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
};

