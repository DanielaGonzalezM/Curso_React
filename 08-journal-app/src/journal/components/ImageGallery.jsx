import React from "react";

export const ImageGallery = () => {
  return <div>ImageGallery</div>;
};
