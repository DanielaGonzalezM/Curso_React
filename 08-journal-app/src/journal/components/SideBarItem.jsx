import { TurnedInNot } from "@mui/icons-material";
import {
  Grid,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from "@mui/material";
import { useMemo } from "react";

export const SideBarItem = ({ title, body, id }) => {
  const newTitle = useMemo(() => {
    return title.length > 17 ? title.substring(0, 17) + "..." : title;
  }, [title]);
  return (
    <ListItem disablePadding>
      <ListItemButton>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>
        <Grid container>
          <ListItemText primary={newTitle} />
          <ListItemText secondary={body} />
        </Grid>
      </ListItemButton>
    </ListItem>
  );
};
