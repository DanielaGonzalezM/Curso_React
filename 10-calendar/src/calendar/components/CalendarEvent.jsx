export const CalendarEvent = ({ event }) => {
  const { title, user } = event;
  return (
    <>
      <strong>{title}</strong>
      <span> - {user.name}</span>
    </>
  );
};
