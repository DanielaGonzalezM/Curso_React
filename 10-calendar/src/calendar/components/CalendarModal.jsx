import { addHours } from "date-fns";
import { useState } from "react";
import Modal from "react-modal";
import DatePicker, { registerLocale } from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import es from "date-fns/locale/es";
registerLocale("es", es);

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
  },
};
Modal.setAppElement("#root");
export const CalendarModal = () => {
  const [isOpen, setIsOpen] = useState(true);
  const [formValues, setFormValues] = useState({
    title: "Daniela",
    notes: "test",
    start: new Date(),
    end: addHours(new Date(), 2),
  });

  const onInputChanged = ({ target }) => {
    setFormValues({ ...formValues, [target.name]: target.value });
  };
  const onDateChanged = (event, changing) => {
    setFormValues({ ...formValues, [changing]: event });
  };
  const onCloseModal = () => {
    setIsOpen(false);
  };
  return (
    <Modal
      isOpen={isOpen}
      onRequestClose={onCloseModal}
      style={customStyles}
      className="modal"
      overlayClassName={"modal-fondo"}
      closeTimeoutMS={200}
    >
      <h1> Nuevo evento </h1>
      <hr />
      <form className="container">
        <div className="form-group mb-2">
          <label>Fecha y hora inicio</label>
          <DatePicker
            selected={formValues.start}
            className="form-control"
            onChange={(event) => {
              onDateChanged(event, "start");
            }}
            dateFormat="Pp"
            showTimeSelect
            locale="es"
            timeCaption="Hora"
          />
        </div>

        <div className="form-group mb-2">
          <label>Fecha y hora fin</label>
          <DatePicker
            minDate={formValues.start}
            selected={formValues.end}
            className="form-control"
            onChange={(event) => {
              onDateChanged(event, "end");
            }}
            dateFormat="Pp"
            showTimeSelect
            locale="es"
            timeCaption="Hora"
          />
        </div>

        <hr />
        <div className="form-group mb-2">
          <label>Titulo y notas</label>
          <input
            type="text"
            className="form-control"
            placeholder="Título del evento"
            name="title"
            autoComplete="off"
            value={formValues.title}
            onChange={onInputChanged}
          />
          <small id="emailHelp" className="form-text text-muted">
            Una descripción corta
          </small>
        </div>

        <div className="form-group mb-2">
          <textarea
            type="text"
            className="form-control"
            placeholder="Notas"
            rows="5"
            name="notes"
            value={formValues.notes}
            onChange={onInputChanged}
          ></textarea>
          <small id="emailHelp" className="form-text text-muted">
            Información adicional
          </small>
        </div>

        <button type="submit" className="btn btn-outline-primary btn-block">
          <i className="far fa-save"></i>
          <span> Guardar</span>
        </button>
      </form>
    </Modal>
  );
};
