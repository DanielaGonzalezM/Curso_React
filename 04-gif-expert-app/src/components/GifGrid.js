import React from 'react'
import { useFetchGift } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    
    const {data:images,loading} = useFetchGift(category);

    return (
        <>
            <h3>{category}</h3>
            {loading&&<p className='animate__animated animate__flash'>loading</p>}
            <div className='card-grid'>
                {images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img} />
                ))}

            </div>
        </>
    )
}
