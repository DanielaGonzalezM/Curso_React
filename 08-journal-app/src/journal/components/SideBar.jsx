import { Box, Divider, Drawer, List, Toolbar, Typography } from "@mui/material";
import { useSelector } from "react-redux";
import { SideBarItem } from "./SideBarItem";
export const SideBar = ({ drawerWidth = 240 }) => {
  const { displayName } = useSelector((state) => state.auth);
  const { notes } = useSelector((state) => state.journal);
  return (
    <Box
      component="nav"
      sx={{ width: { sm: `${drawerWidth}px` }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="permanent"
        open
        sx={{
          display: { xs: "block" },
          "& .MuiDrawer-paper": {
            boxSizing: "border-box",
            width: `${drawerWidth}px`,
          },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />
        <List>
          {notes.map((note) => (
            <SideBarItem key={note.id} {...note} />
          ))}
        </List>
      </Drawer>
    </Box>
  );
};
