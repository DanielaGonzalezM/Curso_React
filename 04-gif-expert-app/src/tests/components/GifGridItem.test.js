import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas componente GifGridItem',()=>{

    const img={
        title:'cat',
        url:'https://giphy.com/gifs/emote-catjam-jpbnoe3UIa8TU8LM13'
    }
    const wrapper=shallow(<GifGridItem
        key={img.id}
        {...img} />);
    test('El componente GifGridItem debe mostrarse correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
     });

     test('debe tener  un pÃ¡rrafo con el title', () => { 
        const p= wrapper.find('p');
        expect(p.text().trim()).toBe(img.title); 
      });
      test('debe tener la imagen url', () => { 
        const imge= wrapper.find('img');
        expect(imge.prop('src')).toBe(img.url); 
      });

      test('debe tener animate__fadeIn', () => { 
        const div= wrapper.find('div');
        const className=div.prop('className');
        expect(className.includes('animate__zoomIn')).toBe(true); 
      });
});