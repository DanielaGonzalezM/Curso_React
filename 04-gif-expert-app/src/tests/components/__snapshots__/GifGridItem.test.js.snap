// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente GifGridItem El componente GifGridItem debe mostrarse correctamente 1`] = `
<div
  className="card  animate__animated  animate__zoomIn"
>
  <img
    alt="cat"
    src="https://giphy.com/gifs/emote-catjam-jpbnoe3UIa8TU8LM13"
  />
  <p>
    cat
  </p>
</div>
`;
